src/main/java/bdv/bigcat/viewer/atlas/source/SourceInfo.java

```diff
@@ -173,6 +173,8 @@
 	public synchronized < T > void removeSource( final Source< T > source )
 	{
 		final int currentSourceIndex = this.sources.indexOf( source );
+		if ( this.states.get( source ).typeProperty().get() == TYPE.LABEL )
```

Comment:
Could you replace this `if` by
```
Optional.ofNullable( this.states.get( source ).meshManagerProperty().get() ).ifPresent( MeshManager::removeAllNeurons );
```
Maybe in the future we will have `MeshManagers` for non-label sources.